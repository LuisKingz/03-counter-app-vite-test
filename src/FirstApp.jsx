import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {    
    return (
    <>
        <h2 data-testid="test-title" >{ title }</h2>
        <p>{ subTitle }</p>
        <p> { name } </p>
    </>
  )
}

// RESTRICCIONES PARA EL TIPO DE DATOS QUE SE RECIBE EN LAS PROPS

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

//  VALORES POR DEFECTO EN CASO DE QUE NO SE ENVIEN LOS PROPS. ENTRAN PRIMERO QUE LOS PROPTYPES
FirstApp.defaultProps = {
    name: "No hay nombre",
    subTitle:"No hay Subtitulo"
}