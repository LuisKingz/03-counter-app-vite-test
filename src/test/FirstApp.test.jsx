import { render } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en FirstApp', () =>{
    test('debe hacer match con el snapshot', () => { 

        const title = "Es un titulo";
        const { container } = render(<FirstApp title={ title }/>);
        expect(container).toMatchSnapshot()

     });

     test('Debe de mostrar el titulo en un h1', () => {         
        const title = "Es un titulo";
        const { container,getByText,getByTestId } = render(<FirstApp title={ title }/>); 
        expect(getByTestId('test-title').innerHTML).toBe(title);
        // expect( getByText(title) ).toBeTruthy();
    });

    test('Debe de mostrar el subtitulo pasado por props', () => { 
        const title = "Es un titulo";
        const subtitulo = "Es un subtitulo";
        const { getByText, } = render(<FirstApp title={ title } subTitle={ subtitulo}/>); 
        expect(getByText(subtitulo)).toBeTruthy();

     })
})