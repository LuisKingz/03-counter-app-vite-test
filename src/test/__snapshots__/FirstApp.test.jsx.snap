// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Es un titulo
  </div>
  <p>
    No hay Subtitulo
  </p>
  <p>
     
    No hay nombre
     
  </p>
</div>
`;
